internal/encryption/targets_test.go

```diff
@@ -175,6 +175,62 @@ func TestBaseEncryption_buildTargetMethods(t *testing.T) {
 			`,
 			wantErr: "Test Config Source:3,12-34: Invalid Key Provider expression format; Expected key_provider.<type>.<name>",
 		},
+		"unused-key-provider": {
```

Comment:
`suggestion`
Could we add a unit test similar to [this one](https://github.com/opentofu/opentofu/pull/2547/files#diff-89663d138a7de3a7669ab0efd13ac851af80d5f7a6a4179b6e8e250682f5ee1bR209)?

Reply:
I've cherry picked in the tests from that commit, with your authorship :+1: